gst/droidcamsrc/gstdroidcamsrc.c

```diff
@@ -354,24 +355,25 @@ gst_droidcamsrc_set_property (GObject * object, guint prop_id,
       g_mutex_unlock (&src->capture_lock);
 
       /* deactivate old mode */
-      if (src->active_mode) {
+      was_active = (src->active_mode != NULL);
+      if (was_active) {
         gst_droidcamsrc_mode_deactivate (src->active_mode);
         src->active_mode = NULL;
       }
 
       src->mode = mode;
 
-      g_rec_mutex_lock (&src->dev_lock);
+      if (was_active) {
```

Comment:
Doesn't this mean that the mode isn't set unless there's already an active mode? If that's the case, then how does the mode get initially set?

Reply:
Active "mode" is set as part of `PAUSED -> PLAYING` transition and is unset as part of `PLAYING -> PAUSED` transition. That is, there's no active "mode" while the element is not `PLAYING`.

Reply:
Ah I'd missed a second call to gst_droidcamsrc_select_and_activate_mode for GST_STATE_CHANGE_PAUSED_TO_PLAYING. That's fine then.